lib/Conversion/AffineToPipeline/AffineToPipeline.cpp

```diff
@@ -447,32 +534,24 @@ LogicalResult AffineToPipeline::createPipelinePipeline(
     auto *stageTerminator = stageBlock.getTerminator();
     builder.setInsertionPointToStart(&stageBlock);
 
-    // Sort the group according to original dominance.
-    llvm::sort(group,
-               [&](Operation *a, Operation *b) { return dom.dominates(a, b); });
-
-    // Move over the operations and add their results to the terminator.
-    SmallVector<std::tuple<Operation *, Operation *, unsigned>> movedOps;
     for (auto *op : group) {
-      unsigned resultIndex = stageTerminator->getNumOperands();
-      auto *newOp = builder.clone(*op, valueMap);
-      if (opsWithReturns.contains(op)) {
-        stageTerminator->insertOperands(resultIndex, newOp->getResults());
-        movedOps.emplace_back(op, newOp, resultIndex);
-      }
+      auto *newOp = builder.clone(*op, valueMaps.data()[startTime]);
+
+      for (auto result : op->getResults())
+        valueMaps.data()[startTime].map(
+            result, newOp->getResult(result.getResultNumber()));
     }
 
-    // Add the stage results to the value map for the original op.
-    for (auto tuple : movedOps) {
-      Operation *op = std::get<0>(tuple);
-      Operation *newOp = std::get<1>(tuple);
-      unsigned resultIndex = std::get<2>(tuple);
-      for (size_t i = 0; i < newOp->getNumResults(); ++i) {
-        auto newValue = stage->getResult(resultIndex + i);
-        auto oldValue = op->getResult(i);
-        valueMap.map(oldValue, newValue);
-      }
+    SmallVector<Value> stageOperands;
```

Comment:
This block of code is pretty dense, please add a comment explaining what it does.